web-common/src/features/file-explorer/new-files.ts

```diff
@@ -9,11 +12,10 @@ import { get } from "svelte/store";
 export async function handleEntityCreate(kind: ResourceKind) {
   if (!(kind in ResourceKindMap)) return;
   const instanceId = get(runtime).instanceId;
-  const allNames = await fetchAllFileNames(queryClient, instanceId);
-  const { name, folder, baseContent, extension } = ResourceKindMap[kind];
+  const allNames = await fetchAllFileNames(queryClient, instanceId, false);
+  const { name, extension, baseContent } = ResourceKindMap[kind];
   const newName = getName(name, allNames);
-
-  const newPath = `${folder ?? name + "s"}/${newName}${extension ?? ".yaml"}`;
+  const newPath = `${name + "s"}/${newName}${extension}`;
 
   await runtimeServicePutFile(instanceId, newPath, {
```

Comment:
**Scalability Issue**: The function handleEntityCreate fetches all file names without considering the scalability impact of fetching large datasets, especially in systems with a vast number of files. <br> **Fix**: Cache file names or use a more efficient querying mechanism that can handle large datasets without significant performance degradation. Consider lazy loading or paginating file name fetches. <br> **Code Suggestion**: 
 ```
 +  const allNames = await fetchAllFileNames(queryClient, instanceId, { pagination: true });
 +  // Implement caching mechanism for file names to improve scalability
 ```